modules/core/src/lib/attribute/data-column.js

```diff
@@ -99,6 +100,11 @@ export default class DataColumn {
     return this._buffer;
   }
 
+  get byteOffset() {
+    const {offset} = this.settings;
+    return this.doublePrecision && this.value instanceof Float64Array ? offset * 2 : offset;
```

Comment:
Does this need to be dynamic? Seems like it could be cached on updates to offset or value.

Reply:
This getter is fairly cheap and infrequently accessed (only on attribute updates). Because there are multiple ways to update data, reducing the number of properties that need maintenance will help control the complexity of the class.